addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
But you don't get rid of that now, first the calculation is done and then you pass it to the big decimal, you don't use big decimal for calculations only for storage.